x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/rollup/v2/TransportRollupAction.java

```diff
@@ -121,9 +123,22 @@ protected void masterOperation(Task task, RollupAction.Request request, ClusterS
             .indices(originalIndexName)
             .fields(request.getRollupConfig().getAllFields().toArray(new String[0]));
         fieldCapsRequest.setParentTask(clusterService.localNode().getId(), task.getId());
-        CreateIndexRequest req = new CreateIndexRequest(tmpIndexName, Settings.builder()
-            .put(IndexMetadata.SETTING_INDEX_HIDDEN, true).build())
-            .mapping("_doc", mapping);
+        // Add the source index name and UUID to the rollup index metadata. If the original index is a rollup index itself,
+        // we will add the name and UUID of the raw index that we initially rolled up.
+        IndexMetadata originalIndexMetadata = state.getMetadata().index(originalIndexName);
+        String sourceIndexName = IndexMetadata.INDEX_ROLLUP_SOURCE_NAME.exists(originalIndexMetadata.getSettings()) ?
+            IndexMetadata.INDEX_ROLLUP_SOURCE_NAME.get(originalIndexMetadata.getSettings()) : originalIndexName;
+        String sourceIndexUuid = IndexMetadata.INDEX_ROLLUP_SOURCE_UUID.exists(originalIndexMetadata.getSettings()) ?
+            IndexMetadata.INDEX_ROLLUP_SOURCE_UUID.get(originalIndexMetadata.getSettings()) : originalIndexMetadata.getIndexUUID();
+
+        // Using the CreateIndexRequest class only to produce the transformation to the mappings map.
+        // Alternatively, we would have to copy-paste that part of the code.
+        Map<String, String> mappingAsMap = new CreateIndexRequest().mapping("_doc", mapping).mappings();
```

Comment:
@talevy Do you have any idea if there's a better way to create the mappings map?

Reply:
```
XContentHelper.convertToMap(XContentType.JSON.xContent(), Strings.toString(mapping), false)
```

should get the `Map<String, Object>`

or

```
XContentParser parser = null;
Map<String, String> mappingAsMap = null;
try {
  parser = xContent.createParser(NamedXContentRegistry.EMPTY, DeprecationHandler.THROW_UNSUPPORTED_OPERATION, Strings.toString(mapping))
  mappingAsMap = parser.mapStrings();
} finally {
  if (parser != null) {
    parser.close();
  }
}
```

I don't know if that is the best way, but the XContentParser#mapStrings looks promising. I wonder what the strategy for changing things between 7.x to 8.0 were here. that technique can likely be leveraged.

Reply:
I think in the piece of code that I shared a few weeks ago, I just re-used the `CreateIndexRequest` as you did. I felt it was the most readable at the time.